services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/hakemus/Hakemus.kt

```diff
@@ -4,41 +4,92 @@ import fi.hel.haitaton.hanke.allu.ApplicationStatus
 import fi.hel.haitaton.hanke.application.ApplicationArea
 import fi.hel.haitaton.hanke.application.ApplicationType
 import fi.hel.haitaton.hanke.application.PostalAddress
+import fi.hel.haitaton.hanke.domain.HasId
 import java.time.ZonedDateTime
 
 data class Hakemus(
-    val id: Long?,
+    override val id: Long,
     val alluid: Int?,
     val alluStatus: ApplicationStatus?,
     val applicationIdentifier: String?,
     val applicationType: ApplicationType,
     val applicationData: HakemusData,
-    val hankeTunnus: String
-)
+    val hankeTunnus: String,
+    val hankeId: Int,
+) : HasId<Long> {
+    fun toResponse(): HakemusResponse =
+        HakemusResponse(
+            id = id,
+            alluid = alluid,
+            alluStatus = alluStatus,
+            applicationIdentifier = applicationIdentifier,
+            applicationType = applicationType,
+            applicationData = applicationData.toResponse(),
+            hankeTunnus = hankeTunnus,
+        )
+}
 
 sealed interface HakemusData {
+    val applicationType: ApplicationType
     val name: String
+    val pendingOnClient: Boolean
     val postalAddress: PostalAddress?
     val startTime: ZonedDateTime?
     val endTime: ZonedDateTime?
     val areas: List<ApplicationArea>?
     val customerWithContacts: Hakemusyhteystieto?
+
+    fun toResponse(): HakemusDataResponse
+
+    fun yhteystiedot(): List<Hakemusyhteystieto>
 }
 
 data class JohtoselvityshakemusData(
+    override val applicationType: ApplicationType = ApplicationType.CABLE_REPORT,
     override val name: String,
     override val postalAddress: PostalAddress? = null,
-    val constructionWork: Boolean? = null,
-    val maintenanceWork: Boolean? = null,
-    val propertyConnectivity: Boolean? = null,
-    val emergencyWork: Boolean? = null,
+    val area: Double? = null,
```

Comment:
Do we need `area`?

Reply:
Good call. I don't think we need it here. Removed.